packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -214,12 +215,205 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		assert(this.hostPolicy.supportGetSnapshotApi !== true, "api should not be called yet");
-		// This is just temporary as this api is not yet enabled in service policies.
-		return this.fetchSnapshot(
-			this.hostPolicy.snapshotOptions,
-			snapshotFetchOptions?.scenarioName,
+		// If no groupId is specified, then fetch the snapshot with "default" groupId.
+		const groupIds = snapshotFetchOptions?.groupIds ?? [];
+		if (groupIds.length === 0) {
+			groupIds.push(defaultGroupIdForSnapshot);
+		}
+		const { snapshot } = await this.startSnapshotFetch({ ...snapshotFetchOptions, groupIds });
+		return snapshot;
+	}
+
+	private async startSnapshotFetch(
```

Comment:
I'm not sure what "start" means in the name. Is there an "endSnapshotFetch"?

Reply:
Renamed.